lib/fluent/plugin/out_s3.rb

```diff
@@ -272,6 +277,35 @@ def write(chunk)
                    else
                      @time_slice_with_tz.call(metadata.timekey)
                    end
+      
+      bucket_name = nil
+
+      if @s3_bucket =~ /\$\{.*\}/
+        @s3_bucket.scan(/\$\{([^\$\{\}]+)\}/) do |placeholder|
```

Comment:
Time placeholders does not use $ character. 
We can use Validator class instead.
https://github.com/fluent/fluentd/blob/master/lib/fluent/plugin/output.rb#L607

Reply:
```ruby
if !@tag_placeholders.empty? || !@keys_placeholders.empty? || !@time_placeholders.empty?
```
?

Reply:
Tag, time and keys placeholders are scanned on @s3_bucket on star. If they're empty, the bucket is created during the star and anytime else in the future.

On the other way, if placeholders are found in @s3_bucket, it means bucket object must be evaluated, everything time before a buffer flush. This is done inside the write method. In the write method I check this situation evaluating whether @bucket is instantiated or not.